### PR TITLE
GH-328: Add Consumer-Aware Error Handlers

### LINKED ISSUE #328: Add ConsumerAwareKafkaListenerErrorHandler
Enable seek on error. See #323 for discussion.

### PR DESCRIPTION
**Pull request by garyrussell**
Resolves #328

Add consumer-aware versions of error handlers, at the container and `@KafkaListener` level.

### CODE REVIEW

**Comment by garyrussell** [comment]
Added a fix for related issue #330

File: src/reference/asciidoc/kafka.adoc [hunk 1]
```diff
@@ -1036,6 +1052,73 @@ public interface KafkaListenerErrorHandler {
...
+[source, java]
+----
+void handle(Exception thrownException, ConsumerRecord<?, ?> data, Consumer<?, ?> consumer);
+
+void handle(Exception thrownException, ConsumerRecords<?, ?> data, Consumer<?, ?> consumer);
+----
+
+respectively.
+
+Similar to the `@KafkaListener` error handlers, you can reset the offsets as needed based on the data the failed.
```
Comment (artembilan) [inline src/reference/asciidoc/kafka.adoc]: `that failed` ?
Can fix one merge

Comment (garyrussell) [inline src/reference/asciidoc/kafka.adoc]: Yes; thanks.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-kafka/commit/f03367cec270f526ee1c1e1a4ebdb069c2e876a7

**Pull request closed without merging**